src/components/TodoList/TodoItem.jsx

```diff
@@ -1,87 +1,67 @@
 import { useState, useEffect } from "react";
-import { AiOutlineEdit, AiOutlineDelete, MdOutlineDone, TbArrowsUp } from "react-icons/all";
+import { AiOutlineEdit, AiOutlineDelete, MdOutlineDone } from "react-icons/all";
 import PriorityIcon from "./PriorityIcon.jsx";
 import axios from "axios";
+import { todoContainerColor } from "../../common/utils.js";
 
-const TodoItem = (todoItem) => {
-    const todoItemData = todoItem.todoItem;
-    const [gradientColor, setGradientColor] = useState("from-yellow-100");
-    const [borderColor, setBorderColor] = useState("border-black");
+const TodoItem = ({ todoItem = {}, todos = [], setTodos }) => {
+    const priority = todoItem.priority;
+    const emptyField = "N/A";
 
-    const handleDeleteTodo = (todoItem) => {
-        axios
-            .delete(`${import.meta.env.VITE_TODO_API}/${todoItem.id}`)
-            //             .then(setIsLoading(true))
-            .catch((err) => console.log(err));
-    };
-
-    const handleEditTodo = (todoItem) => {
+    const handleEditTodo = () => {
         console.log("Editing ", todoItem);
     };
 
     const toggleCompleteTodo = () => {
         axios
-            .put(`${import.meta.env.VITE_TODO_API}/${todoItemData.id}`, {
-                ...todoItemData,
-                completed: !todoItemData.completed,
+            .put(`${import.meta.env.VITE_TODO_API}/${todoItem.id}`, {
+                ...todoItem,
+                completed: !todoItem.completed,
+            })
+            .then((res) => {
+                setTodos((prev) => prev.map((item) => (item.id === res.id ? res : item)));
             })
             .catch((err) => console.log(err));
     };
 
-    const changeTodoColor = (priority) => {
-        switch (priority) {
-            case "high":
-                setGradientColor("from-red-400");
-                setBorderColor("border-red-500");
-                break;
-            case "medium high":
-                setGradientColor("from-amber-400");
-                setBorderColor("border-amber-500");
-                break;
-            case "medium":
-                setGradientColor("from-yellow-300");
-                setBorderColor("border-yellow-400");
-                break;
-            case "medium low":
-                setGradientColor("from-green-300");
-                setBorderColor("border-green-400");
-                break;
-            case "low":
-                setGradientColor("from-green-500");
-                setBorderColor("border-green-600");
-                break;
-        }
-        if (todoItemData.completed === true) {
-            setGradientColor("from-gray-500");
-            setBorderColor("border-black-500");
-        }
+    const handleDeleteTodo = () => {
+        axios
+            .delete(`${import.meta.env.VITE_TODO_API}/${todoItem.id}`)
+            .then(
+                setTodos((prev) => {
```

Comment:
then should have an anonymous function inside, because right now what happens is `.then((res) => setTodos()(res))` you return a function, that you call with another argument. This is a form currying.
Should be `.then(() => setTodos())`